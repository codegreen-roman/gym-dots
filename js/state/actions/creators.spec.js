import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {
    ping,
    startWorkout,
    setStartedWorkout,
    loadLocationStart,
    startWorkoutWithCountdown,
    loadLocationForUsername
} from './creators'

const mockStore = configureMockStore([thunk])

describe('General Creators', () => {

    describe('.ping', () => {
        it('should return action with a type of PING', () => {
            expect(ping({
                payload: {
                    data: 2
                }
            })).toMatchSnapshot()
        })
    })

    describe('.startWorkout', () => {
        it('should return action with a type of WORKOUT_STATUS', () => {
            expect(startWorkout()).toMatchSnapshot()
        })
    })

    describe('.setStartedWorkout', () => {
        it('should return action with a type of WORKOUT_STATUS', () => {
            expect(setStartedWorkout()).toMatchSnapshot()
        })
    })

    describe('.loadLocationStart', () => {
        it('should return action with a type of GET_LOCATION_START', () => {
            expect(loadLocationStart()).toMatchSnapshot()
        })
    })

    describe('.startWorkoutWithCountdown', () => {

        const store = mockStore({ defaults: {} })

        it('creates 2 WORKOUT_STATUS actions when dispatching', (done) => {

            const promise = store.dispatch(startWorkoutWithCountdown())

            jest.runAllTimers()

            promise.then(() => {
                expect(store.getActions()).toMatchSnapshot()
                done()
            })
        })
    })

    describe('.loadLocationForUsername', () => {

        const store = mockStore({ defaults: {} })

        it('creates GET_LOCATION_START and GET_LOCATION actions when dispatching', (done) => {

            const promise = store.dispatch(loadLocationForUsername())

            jest.runAllTimers()

            promise.then(() => {
                expect(store.getActions()).toMatchSnapshot()
                done()
            })
        })
    })

})
