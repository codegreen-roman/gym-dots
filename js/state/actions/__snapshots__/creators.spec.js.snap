// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Creators .loadLocationForUsername creates 2 WORKOUT_STATUS actions when loading dispatching 1`] = `
Array [
  Object {
    "payload": Object {
      "status": "started",
    },
    "type": "GET_LOCATION_START",
  },
  Object {
    "payload": Object {
      "location": "Tallinn, Estonia",
      "status": "ready",
    },
    "type": "GET_LOCATION",
  },
]
`;

exports[`General Creators .loadLocationForUsername creates GET_LOCATION_START and GET_LOCATION actions when dispatching 1`] = `
Array [
  Object {
    "payload": Object {
      "status": "started",
    },
    "type": "GET_LOCATION_START",
  },
  Object {
    "payload": Object {
      "location": "Tallinn, Estonia",
      "status": "ready",
    },
    "type": "GET_LOCATION",
  },
]
`;

exports[`General Creators .loadLocationStart should return action with a type of GET_LOCATION_START 1`] = `
Object {
  "payload": Object {
    "status": "started",
  },
  "type": "GET_LOCATION_START",
}
`;

exports[`General Creators .ping should return action with a type of PING 1`] = `
Object {
  "payload": Object {
    "payload": Object {
      "data": 2,
    },
  },
  "type": "PING",
}
`;

exports[`General Creators .setStartedWorkout should return action with a type of WORKOUT_STATUS 1`] = `
Object {
  "payload": Object {
    "status": "started",
  },
  "type": "WORKOUT_STATUS",
}
`;

exports[`General Creators .startWorkout should return action with a type of WORKOUT_STATUS 1`] = `
Object {
  "payload": Object {
    "status": "starting",
  },
  "type": "WORKOUT_STATUS",
}
`;

exports[`General Creators .startWorkoutWithCountdown creates 2 WORKOUT_STATUS actions when dispatching 1`] = `
Array [
  Object {
    "payload": Object {
      "status": "starting",
    },
    "type": "WORKOUT_STATUS",
  },
  Object {
    "payload": Object {
      "status": "started",
    },
    "type": "WORKOUT_STATUS",
  },
]
`;

exports[`General Creators .startWorkoutWithCountdown creates 2 WORKOUT_STATUS actions when loading dispatching 1`] = `
Array [
  Object {
    "payload": Object {
      "status": "starting",
    },
    "type": "WORKOUT_STATUS",
  },
  Object {
    "payload": Object {
      "status": "started",
    },
    "type": "WORKOUT_STATUS",
  },
]
`;
